packages/manifest-helper/src/manifest.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable functional/no-throw-statement */
+import { existsSync, readFileSync } from 'fs'
+import { resolve } from 'path'
+import { getErrorMessage } from './utils/error'
+
+export const MANIFEST_FILE_NAME = 'field-plugin.config.json'
+export const MANIFEST_FULL_PATH = resolve(process.cwd(), MANIFEST_FILE_NAME)
+
+export type ManifestOption = {
+  name: string
+  value: string
+}
+
+export type Manifest = {
+  options: ManifestOption[]
+}
+
+export const manifestExists = (): boolean => {
+  return existsSync(MANIFEST_FULL_PATH)
+}
+
+export const load = (): Manifest => {
```

Comment:
What if we check if the manifest file exists within this `load` function, instead of exposing `manifestExists` function and let the caller handle it?

Reply:
It was my first approach back on time, but It seemed to me that the `load` function would have more than just one responsibility. 

Also, as this function raises an exception, I found it better to give the developer an option to check it before raising an unnecessary exception. 

The approach was similar to the one used by [.NET](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.readallbytes?view=net-7.0#system-io-file-readallbytes(system-string)) to deal with IO operations.
- Provide a method for checking if a file exists
- Provide a method for reading the file and if it does not exist, raise an exception.

What do you think, does it make sense? 🤣

Reply:
Thanks for going deep on this, @eunjae-lee. 🙌 

> Just like that, our manifest helper is literally a helper, doing things for us. So its load method can
>
> - return a manifest if it exists
> - return or throw an error if it doesn't exist
> - return or throw an error if it is invalid

It sounds good to me... If I understood correctly, the code already works like that.

Reply:
@demetriusfeijoo Oh, what I meant was to suggest to include `existsSync(MANIFEST_FULL_PATH)` inside `load`, and not to use `manifestExists` at `deploy/helper.ts`.

Reply:
@eunjae-lee, and throw an exception if it doesn't exist or just return undefined?

Reply:
@demetriusfeijoo it depends. `manifest.ts` in manifest-helper could return something like:

```
{
  type: 'success' | 'not-existing' | 'invalid',
  ..
}
```

However, you already throw an exception when it's invalid including a proper error message, and display the message in `deploy/helper.ts`. So in this case, we can overall use "throw" to handle error situations. However, absence of manifest file here is not an error, and we don't need to show any error message. So I think we can just return undefined. 

What do you think?

Reply:
I think returning undefined will be better if we're not going to let the developer check the existence of the file.

You mean something like [this](https://github.com/storyblok/field-plugin/pull/278/commits/9fa730ac693b47b8840461c544eebb983289fe7c), right